src/components/OnboardingManager.tsx

```diff
@@ -1,114 +1,105 @@
+    progress: onboardingProgress,
+    setProgress,
+    pushingData,
+  } = onboarding;
+  const { profileReady, profileData } = profile;
+  const { userId } = auth;
+
+  useEffect(() => {
     if (
-      onboarding.progress < 0 ||
-      onboarding.progress >= onboarding.flow.length ||
-      !auth?.userId ||
-      (profile?.profileReady && profile?.profileData)
+      // edge case (realistically shouldn't happen)
+      onboardingProgress < 0 ||
+      // progress beyond length of flow
+      (onboardingFlow.length !== 0 &&
+        onboardingProgress >= onboardingFlow.length) ||
+      // already onboarded or not signed in
+      (!pushingData && profileReady && (profileData || !userId))
     ) {
-      router.push(CONFIG.homepage);
+      push(CONFIG.homepage);
       return;
     }
 
-    if (pageProgress > onboarding.progress)
-      router.push(`/onboarding/${onboarding.flow[onboarding.progress].url}`);
-  }, [onboarding, profile, router, pageProgress]);
-
-  const advanceProgress = () => {
-    // safeguard
-    if (!onboarding) return;
-    if (pageProgress > onboarding.progress) return;
+    // advance progress if exactly 1 more
+    if (pageProgress === onboardingProgress + 1) setProgress(pageProgress);
+    // otherwise, rebound to current progress
+    else if (pageProgress > onboardingProgress)
+      push(`/onboarding/${onboardingFlow[onboardingProgress].url}`);
+  }, [
+    onboardingProgress,
+    onboardingFlow,
+    profileReady,
+    profileData,
+    push,
+    pageProgress,
+    userId,
+    setProgress,
+    pushingData,
+  ]);
 
-    const newProgress = pageProgress + 1;
-
-    if (
-      newProgress > onboarding.progress &&
-      onboarding.progress < onboarding.flow.length - 1
-    ) {
-      onboarding.setCanContinue(false);
-      onboarding.setProgress(newProgress);
+  const goToHomepage = () => {
+    if (onboardingProgress === 0) {
+      push(CONFIG.homepage);
+      return;
     }
 
-    if (newProgress >= onboarding.flow.length) {
-      onboarding.flushData().then(() => {
-        router.push(CONFIG.homepage);
-      });
-    } else {
-      router.push(`/onboarding/${onboarding.flow[newProgress].url}`);
-    }
+    // eslint-disable-next-line no-alert
+    const ans = window.confirm(
+      'Are you sure you want to continue? Your changes will not be saved after you leave.',
+    );
+    if (ans) push(CONFIG.homepage);
   };
 
   return (
-    <>
-      <BackLink
-        href={
-          onboarding && pageProgress > 0
-            ? `/onboarding/${onboarding.flow[pageProgress - 1].url}`
-            : CONFIG.homepage
-        }
-      >
-        Back
-      </BackLink>
-      <OuterDiv>
-        <ProgressBar
-          steps={
-            new Set(onboarding ? onboarding.flow.slice(1).map(f => f.name) : [])
-          }
-          progress={pageProgress}
+    <OuterDiv>
+      <UnstyledButton onClick={goToHomepage}>
+        <LogoImage
+          $show={pageProgress > 0}
+          width="205"
```

Comment:
this number 205 comes up a lot, maybe just make this logoimage width default 205? idk though

Reply:
it literally shows up twice david

Reply:
also it would be a hassle to update the actual image file every time, better to be able to downsize